fjord/alerts/api_views.py

```diff
@@ -189,26 +189,29 @@ def post(self, request):
         try:
             flavorslug = request.DATA['flavor']
         except KeyError:
-            return self.rest_error(
-                status=404,
-                errors='Flavor not specified in payload.'
-            )
+            raise exceptions.ParseError({
+                'flavor': [
+                    'Flavor not specified in payload'
+                ]
+            })
 
         try:
             flavor = AlertFlavor.objects.get(slug=flavorslug)
```

Comment:
Are you duplicating the work to get and validate the flavor here? This seems the same as the work done in the validator. Plus, this version doesn't deal with comma separated values?